pkg/containerimagebuilderpusher/kaniko.go

```diff
@@ -300,6 +290,26 @@ func (k *Kaniko) getKanikoJobSpec(namespace string, buildOptions *BuildOptions,
 	return kanikoJobSpec
 }
 
+func (k *Kaniko) getJobName(image string) string {
+
+	// Valid job name is composed from a DNS-1123 subdomains which in turn must contain only lower case
+	// alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com',
+	// regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*
+
+	functionName := strings.Replace(image, "/", "", -1)
```

Comment:
For readability, we can simply take the image name itself using split on `/` and `:`

and regardless, worth using regex test, if it fails, simply fallback to `uuid4` instead of failing the entire process

Reply:
Not sure how splitting is better but added regex based fallback